internal/fleetdb/component.go

```diff
@@ -0,0 +1,92 @@
+	Slot                         string               `json:"slot,omitempty"`
+	Architecture                 string               `json:"architecture,omitempty"`
+	MacAddress                   string               `json:"macaddress,omitempty"`
+	SupportedControllerProtocols string               `json:"supported_controller_protocol,omitempty"`
+	SupportedDeviceProtocols     string               `json:"supported_device_protocol,omitempty"`
+	SupportedRAIDTypes           string               `json:"supported_raid_types,omitempty"`
+	PhysicalID                   string               `json:"physid,omitempty"`
+	FormFactor                   string               `json:"form_factor,omitempty"`
+	PartNumber                   string               `json:"part_number,omitempty"`
+	OemID                        string               `json:"oem_id,omitempty"`
+	DriveType                    string               `json:"drive_type,omitempty"`
+	StorageController            string               `json:"storage_controller,omitempty"`
+	BusInfo                      string               `json:"bus_info,omitempty"`
+	WWN                          string               `json:"wwn,omitempty"`
+	Protocol                     string               `json:"protocol,omitempty"`
+	SmartStatus                  string               `json:"smart_status,omitempty"`
+	SmartErrors                  []string             `json:"smart_errors,omitempty"`
+	PowerCapacityWatts           int64                `json:"power_capacity_watts,omitempty"`
+	SizeBytes                    int64                `json:"size_bytes,omitempty"`
+	CapacityBytes                int64                `json:"capacity_bytes,omitempty" diff:"immutable"`
+	ClockSpeedHz                 int64                `json:"clock_speed_hz,omitempty"`
+	Cores                        int                  `json:"cores,omitempty"`
+	Threads                      int                  `json:"threads,omitempty"`
+	SpeedBits                    int64                `json:"speed_bits,omitempty"`
+	SpeedGbps                    int64                `json:"speed_gbps,omitempty"`
+	BlockSizeBytes               int64                `json:"block_size_bytes,omitempty"`
+	CapableSpeedGbps             int64                `json:"capable_speed_gbps,omitempty"`
+	NegotiatedSpeedGbps          int64                `json:"negotiated_speed_gbps,omitempty"`
+	Oem                          bool                 `json:"oem,omitempty"`
+}
+
+// toComponentSlice converts a common.Device object to the fleetdb component slice object
+func toComponentSlice(serverID uuid.UUID, inventory *common.Device) ([]*fleetdbapi.ServerComponent, error) {
+	componentsTmp := []*fleetdbapi.ServerComponent{}
+
+	// vendor := inventory.Vendor
+	// componentsTmp = append(componentsTmp,
+	// 	bios(vendor, inventory.BIOS),
+	// 	bmc(vendor, inventory.BMC),
+	// 	mainboard(vendor, inventory.Mainboard),
+	// )
+
+	// componentsTmp = append(componentsTmp, dimms(vendor, inventory.Memory)...)
```

Comment:
For line 61-69
Do we still need these like we had for alloy https://github.com/metal-toolbox/alloy/blob/main/internal/store/serverservice/components.go#L64

@DoctorVin